www/js/lib/uiUtil.js

```diff
@@ -20,27 +20,65 @@
  * along with Kiwix (file LICENSE-GPLv3.txt).  If not, see <http://www.gnu.org/licenses/>
  */
 'use strict';
-define([], function() {
 
+// DEV: Put your RequireJS definition in the rqDef array below, and any function exports in the function parenthesis of the define statement
+// (This method is needed in order to load WebP polyfills conditionally)
+var rqDef = [];
+
+// Add WebP polyfill only if webpHero was loaded in init.js
+if (webpMachine) {
+    rqDef.push('webpHeroBundle');
+}
+
+define(rqDef, function() {
+
+    /**
+     * A queue for WebP images to be decoded by the single-threaded WebpMachine
+     * @type Array
+     */
+    var webpQueue = [];
     
     /**
-     * Creates a Blob from the given content, then a URL from this Blob
-     * And put this URL in the attribute of the DOM node
+     * Creates a BLOB from the given content, then a blob: or data: URI from this BLOB
+     * The given attribute of the DOM node (nodeAttribute) is then set to this URI
      * 
      * This is useful to inject images (and other dependencies) inside an article
      * 
-     * @param {Object} jQueryNode
-     * @param {String} nodeAttribute
-     * @param {Uint8Array} content
-     * @param {String} mimeType
+     * @param {Object} node The node to which the BLOB data should be added
+     * @param {String} nodeAttribute The attribute to set to the BLOB URI
+     * @param {Uint8Array} content The binary content to convert to a BLOB URI
+     * @param {String} mimeType The MIME type of the content
      */
-    function feedNodeWithBlob(jQueryNode, nodeAttribute, content, mimeType) {
-        var blob = new Blob([content], {type: mimeType});
-        var url = URL.createObjectURL(blob);
-        jQueryNode.on('load', function () {
-            URL.revokeObjectURL(url);
-        });
-        jQueryNode.attr(nodeAttribute, url);
+    function feedNodeWithBlob(node, nodeAttribute, content, mimeType) {
+        // Decode WebP data if the browser does not support WebP and the mimeType is webp
+        if (webpMachine && /image\/webp/i.test(mimeType)) {
+            // Queue WebP images to be decoded (the WebP polyfill is single-threaded and will reject a job if it is busy)
+            webpQueue.push({ 'node': node, 'nodeAttribute': nodeAttribute, 'content': content });
```

Comment:
We should consider the possibility of applying an image queue in `app.js` to the whole function `loadImagesJQuery()` instead of doing it just for WebP images. The same logic as here could be transferred there very easily. The main advantage would be that images would be loaded from the top of the page, sequentially, rather than in a somewhat random order as at present, at least in jQuery mode (and some browsers do this for SW mode automatically). This would be a better user experience, as images above the fold would appear fairly instantly.

Reply:
If you want.
It's true that it would probably improve the user experience in jQuery mode.
But maybe it should be a separate issue/PR?

Reply:
I'll add it as a separate issue.